config/quality/checkstyle/checkstyle.xml

```diff
@@ -1,238 +1,240 @@
+        <!-- <module name="ImportOrder"/> -->
+        <module name="RedundantImport" />
+        <module name="UnusedImports" />
 
         <!-- Checks for Javadoc comments.                     -->
         <!-- See http://checkstyle.sourceforge.net/config_javadoc.html -->
         <module name="JavadocMethod">
             <property name="scope" value="package" />
-            <property name="allowMissingParamTags" value="true" />
+            <property name="allowMissingPropertyJavadoc" value="true" />
             <property name="allowMissingThrowsTags" value="true" />
-            <property name="allowMissingReturnTag" value="true" />
             <property name="allowThrowsTagsForSubclasses" value="true" />
             <property name="allowUndeclaredRTE" value="true" />
-            <property name="allowMissingPropertyJavadoc" value="true" />
-            <property name="severity" value="ignore" />
         </module>
         <module name="JavadocType">
             <property name="scope" value="package" />
-            <property name="severity" value="ignore" />
         </module>
+        <!-- We ignore this warning, because variable should be self-explanatory, thus eliminating the need for Javadoc -->
         <module name="JavadocVariable">
-            <property name="scope" value="package" />
             <property name="severity" value="ignore" />
         </module>
         <module name="JavadocStyle">
-            <property name="checkEmptyJavadoc" value="true" />
-            <property name="severity" value="ignore" />
+            <property name="checkFirstSentence" value="false" />
         </module>
 
+        <!-- Checks for Metrics. -->
+        <!-- See http://checkstyle.sourceforge.net/config_metrics.html -->
+        <module name="BooleanExpressionComplexity" />
+        <module name="ClassDataAbstractionCoupling" />
+        <module name="ClassFanOutComplexity" />
+        <module name="CyclomaticComplexity" />
+        <module name="JavaNCSS" />
+        <module name="NPathComplexity" />
+
+
+        <!-- Miscellaneous other checks.                   -->
+        <!-- See http://checkstyle.sourceforge.net/config_misc.html -->
+        <module name="ArrayTypeStyle" />
+        <module name="AvoidEscapedUnicodeCharacters" />
+        <module name="CommentsIndentation" />
+        <module name="DescendantToken" />
+        <module name="FileContentsHolder" />
+        <module name="FinalParameters" />
+        <module name="Indentation" />
+        <module name="OuterTypeFilename" />
+        <module name="TodoComment">
+            <property name="format" value="(?i)\s+CR\s+" />
+            <property name="severity" value="info" />
+        </module>
+        <module name="TodoComment">
+            <property name="format" value="(?i)\s+TODO\s+" />
+            <property name="severity" value="info" />
+        </module>
+        <module name="TrailingComment" />
```

Comment:
Hmm, Checkstyle gave pretty valid reasons for it IMO:
"The comments have to be aligned so that they do not interfere with the visual structure of the code. If you don't align them neatly, they'll make your listing look like it's been through a washing machine."
"Endline comments tend to be hard to format...It takes time to align them. Such time is not spent learning more about the code; it's dedicated solely to the tedious task of pressing the spacebar or tab key."
"Endline comments are also hard to maintain. If the code on any line containing an endline comment grows, it bumps the comment farther out, and all the other endline comments will have to bumped out to match. Styles that are hard to maintain aren't maintained...."
"Endline comments also tend to be cryptic. The right side of the line doesn't offer much room and the desire to keep the comment on one line means the comment must be short. Work then goes into making the line as short as possible instead of as clear as possible. The comment usually ends up as cryptic as possible...."
"A systemic problem with endline comments is that it's hard to write a meaningful comment for one line of code. Most endline comments just repeat the line of code, which hurts more than it helps."